tap_github/__init__.py

```diff
@@ -2418,17 +2419,18 @@ def get_all_commit_files(schemas, repo_path,  state, mdata, start_date, gitLocal
                 #logger.info('Head already fetched {} {}'.format(headRef, headSha))
                 continue
 
-            # Emit the ref record as well
-            refRecord = {
-                'id': '{}/{}'.format(repo_path, headRef),
-                '_sdc_repository': repo_path,
-                'ref': headRef,
-                'sha': headSha
-            }
-            with singer.Transformer() as transformer:
-                rec = transformer.transform(refRecord, schemas['refs'],
-                    metadata=metadata.to_map(mdata))
-            singer.write_record('refs', rec, time_extracted=extraction_time)
+            # Emit the ref record as well (only in regular mode, not commit-only)
+            if not commits_only:
```

Comment:
Should this just check if the refs stream is selected?  If not, don't write these records?